File: Extra - Estudos/1382 - ElementarMeuCaroWatson/Main.java
import java.util.*;

public class Main {

    public static int contarTrocas(int[] arr, int tam){
        int count = 0;
        for(int i = 0 ; i < tam - 1 ; i++){
            int menor = i;
            for(int j = i + 1; j < tam ; j++){
                if(arr[menor] > arr[j]){
                    menor = j;
                }
            }
            if (menor != i) { 
                int tmp = arr[i];
                arr[i] = arr[menor];
                arr[menor] = tmp;
                count++;  
            }
        }
        return count;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int instancias = sc.nextInt();
        
        for(int i = 0; i < instancias ; i++){
            int tamanho = sc.nextInt();
            int[] arr = new int[tamanho];

            for(int j = 0 ; j < tamanho ; j++){
                arr[j] = sc.nextInt();
            }
            int contador = contarTrocas(arr, tamanho);
            System.out.println(contador);
        }
        
        sc.close();
    }
    
}

File: Extra - Estudos/1382 - ElementarMeuCaroWatson/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int contarTrocas(int sequencia[], int tam){
    int count = 0;
    for(int i = 0 ; i < tam -1 ; i++){
        int menor = i;
        for(int j = i + 1 ; j < tam ; j++){
            if(sequencia[menor] > sequencia[j]){
                menor = j;
            }
        }
        if(menor != i){
            int tmp = sequencia[i];
            sequencia[i] = sequencia[menor];
            sequencia[menor] = tmp;
            count++;
        }

    }
    return count;

}

int main(){
    int instancia;
    int tam;
    int sequencia[100];


    scanf("%d", &instancia);

    for(int i = 0 ; i < instancia ; i++){
        scanf("%d", &tam);
        for(int j = 0 ; j < tam ; j++){
            scanf("%d", &sequencia[j]);
        }
        int count = contarTrocas(sequencia, tam);
        printf("%d\n", count);
    }



    return 0;
}